bundles/core/src/main/java/com/adobe/cq/commerce/core/components/models/product/Product.java

```diff
@@ -72,6 +72,17 @@
 
     Boolean isBundleProduct();
 
+    /**
+     * The version 1 of the product component always returns <code>false</code> as it does not support this feature.
+     * The version 2 of the product component does support this feature but it requires a Magento EE instance with
+     * at least version 2.4.2.
+     * 
+     * @return <code>true</code> if the product data contains staged changes, <code>false</code> otherwise.
```

Comment:
We didn't do this from the beginning, but this is a good chance to introduce `* @since com.adobe.cq.commerce.core.components.models.product 3.1.0` annotations, like WCM core components have as well.

Reply:
Done